app/components/friendly_date/index.tsx

```diff
@@ -1,187 +1,81 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import moment from 'moment';
 import React from 'react';
 import {type IntlShape, useIntl} from 'react-intl';
 import {type StyleProp, Text, type TextStyle} from 'react-native';
 
 import {DateTime} from '@constants';
-import {isYesterday} from '@utils/datetime';
 
 const {SECONDS} = DateTime;
 
 type Props = {
     style?: StyleProp<TextStyle>;
-    sourceDate?: number | Date;
-    value: number | Date;
+    value: number;
 };
 
-function FriendlyDate({style, sourceDate, value}: Props) {
+function FriendlyDate({style, value}: Props) {
     const intl = useIntl();
-    const formattedTime = getFriendlyDate(intl, value, sourceDate);
+    const formattedTime = getFriendlyDate(intl, value);
     return (
         <Text style={style}>{formattedTime}</Text>
     );
 }
 
-export function getFriendlyDate(intl: IntlShape, inputDate: number | Date, sourceDate?: number | Date): string {
-    const today = sourceDate ? new Date(sourceDate) : new Date();
-    const date = new Date(inputDate);
-    const difference = (today.getTime() - date.getTime()) / 1000;
-    if (difference < 0) {
-        return getFriendlyDateAfter(intl, -difference, date, today);
-    }
+function getDiff(inputDate: number, unit: moment.unitOfTime.Diff) {
+    const input = new Date(inputDate);
+    const today = new Date();
+
+    const momentA = moment.utc(input.getTime());
+    const momentB = moment.utc(today.getTime());
```

Comment:
Why utc?

Reply:
I was mimicking https://github.com/mattermost/mattermost/blob/54bfb24e4fb5a32809ac17f968bed9c4b803f75e/webapp/channels/src/utils/datetime.ts#L42

Both dates are represented as JS timestamps, which should be timezone agnostic. The webapp code transform them to the "correct timezone", but still, the relative difference is going to be the same in any case.

Not sure if I am missing any corner case, but I feel in this case, the simplicity of the code wins over the potential error we might be showing.

Reply:
What about a tomorrow instead of a time diff? Can it happen?

Reply:
Tomorrow happens when we call `intl.formatRelativeTime(1, 'days')`. Moment doesn't take care of the formatting, it just tell us the relative difference in the appropiate unit.

In order to say tomorrow, at least 24 hours must have passed (if not, we say in X hours). So the current time and the due date are 24 hours apart. No matter the timezone (since both values were represented as timestamps), they are at least 24 hours apart. So moment should correctly return a diff of `+1`.